File: backend/app/schemas/notification.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import NotificationType


class NotificationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    company_id: uuid.UUID
    user_id: uuid.UUID
    type: NotificationType
    title: str
    message: str
    resource_type: str | None
    resource_id: uuid.UUID | None
    is_read: bool
    read_at: datetime | None
    created_at: datetime
    updated_at: datetime


class PaginatedNotificationResponse(BaseModel):
    items: list[NotificationResponse]
    page: int
    page_size: int
    total: int
    pages: int


class UnreadCountResponse(BaseModel):
    count: int

File: frontend/src/types/notification.ts
export type NotificationType =
  | 'TASK_ASSIGNED'
  | 'TASK_STATUS_CHANGED'
  | 'LEAVE_APPROVED'
  | 'LEAVE_REJECTED'
  | 'LEAVE_CANCELLED'
  | 'SYSTEM';

export interface NotificationItem {
  id: string;
  company_id: string;
  user_id: string;
  type: NotificationType;
  title: string;
  message: string;
  resource_type: string | null;
  resource_id: string | null;
  is_read: boolean;
  read_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface PaginatedNotifications {
  items: NotificationItem[];
  page: number;
  page_size: number;
  total: number;
  pages: number;
}

export interface UnreadCountResponse {
  count: number;
}
